crates/hyperqueue/src/dashboard/ui/screens/home/chart_section.rs

```diff
@@ -0,0 +1,132 @@
+use std::time::{Duration, SystemTime, UNIX_EPOCH};
+use tako::messages::gateway::CollectedOverview;
+
+use crate::dashboard::ui::terminal::DashboardFrame;
+use tui::layout::{Alignment, Rect};
+use tui::style::{Color, Modifier, Style};
+use tui::symbols;
+use tui::text::{Span, Spans};
+use tui::widgets::{Axis, Block, Borders, Chart, Dataset, Paragraph, Wrap};
+
+/// A new WorkerTimeState is created everytime there is a #worker change event
+#[derive(Clone)]
+struct WorkerCountAtTime {
+    pub at_time: SystemTime,
+    pub worker_count: u32,
+}
+
+#[derive(Clone)]
+pub struct ClusterOverviewChart {
+    /// The state of the worker at different times.
+    worker_count_vec: Vec<WorkerCountAtTime>,
+    /// The time for which the data is plotted for
+    view_size: Duration,
+}
+
+impl ClusterOverviewChart {
+    pub fn update(&mut self, overview: &CollectedOverview) {
+        self.worker_count_vec.push(WorkerCountAtTime {
+            at_time: SystemTime::now(),
+            worker_count: overview.worker_overviews.len() as u32,
+        });
+    }
+
+    pub fn draw(&mut self, rect: Rect, frame: &mut DashboardFrame) {
+        let max_workers_in_history = self.get_max_workers_in_view().unwrap_or(0);
+        if max_workers_in_history > 0 {
```

Comment:
Fill the worker counts with zeros if there are no records in the time window.

Reply:
Slightly confused about this one, it is 0 because of

 ```    pub fn update(&mut self, overview: &CollectedOverview) {
        self.worker_records.push(WorkerCountRecord {
            at_time: SystemTime::now(),
            worker_count: overview.worker_overviews.len() as u32,
        });
    }
```
perhaps I am missing something

Reply:
The chart could draw 0 on the Y axis for time before the server was started. But it's probably useless.
In any case, this condition should be always true, so remove it and the `else` case. If there are no workers yet, just don't use an empty dataset.